### ISSUE TITLE
Regression when using const Tensor * ?

### ISSUE THREAD

**Issue by ninnghazad** [romeric/Fastor#70]
Just pulled the latest and found i could not compile anymore. 
The problem was when doing something like this:
```
struct Foo {
  Tensor<float,2,2,2> bar;
};
const Foo * foo = ...;

some_other_tensor += foo->bar(fseq<0,2>(),fseq<0,2>(),fall);
```

resulting in an error like this:
```
./submodules/Fastor/Fastor/expressions/views/tensor_fixed_views_nd.h:144:42: error: no matching function for call to ‘Fastor::TensorConstFixedViewExprnD<Fastor::Tensor<float, 16, 16, 4>, Fastor::fseq<0, 16, 1>, Fastor::fseq<14, 16, 1>, Fastor::fseq<0, -1, 1> >::get_index_s<0, 2>::Do(<unresolved overloaded function type>, const std::array<int, 3>&)’
./submodules/Fastor/Fastor/expressions/views/tensor_fixed_views_nd.h:193:21: note: candidate: ‘template<long unsigned int DIMS> static void Fastor::TensorConstFixedViewExprnD<TensorType<T, Rest ...>, Fseqs ...>::get_index_s<from, to>::Do(int&, const std::array<int, DIMS>&) [with long unsigned int DIMS = DIMS; long unsigned int from = 0; long unsigned int to = 2; TensorType = Fastor::Tensor; T = float; long unsigned int ...Rest = {16, 16, 4}; Fseqs = {Fastor::fseq<0, 16, 1>, Fastor::fseq<14, 16, 1>, Fastor::fseq<0, -1, 1>}]’
  193 |         static void Do(int &ind, const std::array<int,DIMS>& as) {
      |                     ^~
./submodules/Fastor/Fastor/expressions/views/tensor_fixed_views_nd.h:193:21: note:   template argument deduction/substitution failed:
./submodules/Fastor/Fastor/expressions/views/tensor_fixed_views_nd.h:144:42: note:   cannot convert ‘Fastor::sum’ (type ‘<unresolved overloaded function type>’) to type ‘int&’
  144 |                 get_index_s<0,DIMS-1>::Do(sum, as);

```

using non-const pointers to the objects containing Tensor members fixed these cases.

**Comment by romeric** [comment]
I have been pushing some new stuff that impacts precisely your usecase (fully multi-dimensional compile time views). Will fix ASAP. 

Thanks for reporting

**Comment by romeric** [comment]
This was an actual bug/typo. Fixed here 3aced3e. Let me know otherwise. Thanks for reporting. 

By the way when creating static views `fall` is the same as `all` so `A(all,fseq<0,2>())` actually creates a static view rather than dynamic. Just less typing.
